src/main/java/net/rptools/maptool/client/ui/zone/FogUtil.java

```diff
@@ -96,10 +96,19 @@ public class FogUtil {
      */
     List<Geometry> visibleAreas = new ArrayList<>();
     final List<Function<VisionBlockingAccumulator, Boolean>> topologyConsumers = new ArrayList<>();
-    topologyConsumers.add(acc -> acc.addWallBlocking(topology));
-    topologyConsumers.add(acc -> acc.addHillBlocking(hillVbl));
-    topologyConsumers.add(acc -> acc.addPitBlocking(pitVbl));
-    topologyConsumers.add(acc -> acc.addCoverBlocking(coverVbl));
+    if (topology != null) {
```

Comment:
No need to support `null` here. Callers should pass an empty tree if no blocking should be applied. But also see my other comment about why the new calls with `null` don't even need to be made.